de-webapp/src/main/java/org/iplantc/de/server/websocket/amqp/AMQPConnectionManager.java

```diff
@@ -56,6 +61,9 @@ public static AMQPConnectionManager getInstance() {
     }
 
     public Connection getConnection() {
+        if (connection == null) {
+            connection = createConnection();
+        }
```

Comment:
I wonder if another call should be made to verify that the connection is good (in addition to verifying that it's not null) might be necessary. For example, the connection might have been lost because of network shenanigans, but the `connection` property still might not be null.

Reply:
That's a good question.  As far as I can recall, when the UI has stopped receiving notifications, the connection has been null so I wasn't too concerned about other cases, but I can look into it.